samples/snap-geometry-edits-with-utility-network-rules/src/main/java/com/esri/arcgismaps/sample/snapgeometryeditswithutilitynetworkrules/components/SnapGeometryEditsWithUtilityNetworkRulesViewModel.kt

```diff
@@ -0,0 +1,514 @@
+        // Set new value into appropriate property via index
+        _snapSourcePropertyList.value[index].snapSourceSettings.isEnabled = checkedValue
+
+        updateSnapSourceList()
+    }
+
+    /**
+     * Updates the lists used by the UI to show SnapSourceSettings information.
+    */
+    private fun updateSnapSourceList() {
+        // Update the backing list with a new list of current properties from snapSettings
+        _snapSourcePropertyList.value = currentSnapSourcePropertyList()
+    }
+
+    /**
+     * Returns a list of [SnapSourceProperty] objects based on the current snap sources.
+     */
+    private fun currentSnapSourcePropertyList(): List<SnapSourceProperty> {
+        return buildList {
+            geometryEditor.snapSettings.sourceSettings.forEach { sourceSettings ->
+                when (sourceSettings.source) {
+                    is GraphicsOverlay -> {
+                        symbolSwatches[sourceSettings.ruleBehavior]?.let { swatch ->
+                            add(SnapSourceProperty((sourceSettings.source as GraphicsOverlay).id, swatch, sourceSettings))
+                        }
+
+                        // Set the appropriate symbol for the layer based on the SnapRuleBehavior.
+                        graphicsOverlay.renderer = SimpleRenderer(symbols[sourceSettings.ruleBehavior])
+                    }
+                    is SubtypeFeatureLayer -> {
+                        sourceSettings.childSourceSettings.forEach { childSourceSettings ->
+(childSourceSettings.source as? SubtypeSublayer)?.let { childSource ->
+                                when (childSource.name) {
+                                    distributionPipeName, servicePipeLayerName -> {
+                                        symbolSwatches[childSourceSettings.ruleBehavior]?.let { swatch ->
+                                            add(SnapSourceProperty(childSource.name, swatch, childSourceSettings))
+                                        }
+
+                                        // Set the appropriate symbol for the sublayer based on the SnapRuleBehavior.
+                                        childSource.renderer =
+                                            SimpleRenderer(symbols[childSourceSettings.ruleBehavior])
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Clears the selection on the layer and reinstates feature visibility, then resets the selected
+     * feature and layer and any UI backing variables.
+     */
+    private fun resetSelections() {
+        // Clear the existing selection and show the selected feature                                      
+        (selectedFeature?.featureTable?.layer as? FeatureLayer)?.let { 
+            it.clearSelection()
+            it.setFeatureVisible(selectedFeature!!, true)
+        }
```

Comment:
Avoid forcing nullability with `!!`: 
```suggestion
        selectedFeature?.let { feature ->
            (feature.featureTable?.layer as? FeatureLayer)?.let {
                it.clearSelection()
                it.setFeatureVisible(feature = feature, visible = true)
            }
        }
```